tsinfer/formats.py

```diff
@@ -74,6 +74,43 @@ def permissive_json_schema():
     }
 
 
+def node_metadata_schema():
+    # This is fixed by tsinfer: users cannot add to the node metadata
+    return tskit.MetadataSchema(
+        {
+            "codec": "struct",
+            "type": ["object", "null"],
+            "properties": {
+                "tsinfer": {
```

Comment:
I don't see the point in the "tsinfer" top key here.

Reply:
I thought it was cleaner, because then when we e.g. add values inferred from `tsdate` we can store them in a `tsdate` property. That way it is clear where the different metadata values have come from, and they don't really risk overwriting each other etc.

Reply:
Addendum: I see this as a nice advantage of `struct`: we can be verbose in the explanation (and nesting) of the properties in the metadata, and it doesn't take up extra space in the encoding of the metadata for each row. That's my understanding, anyway.

Reply:
The nice thing is that there is no storage cost to the additional level.

Reply:
> The nice thing is that there is no storage cost to the additional level.

Yes, that's what I was trying to say (but you said it better)